graspologic/cluster/base.py

```diff
@@ -6,11 +6,12 @@
 
 import numpy as np
 from sklearn.base import BaseEstimator, ClusterMixin
+from sklearn.cluster import AffinityPropagation
 from sklearn.metrics import adjusted_rand_score
 from sklearn.utils.validation import check_is_fitted
 
 
-class BaseCluster(ABC, BaseEstimator, ClusterMixin):
+class BaseCluster(ABC, AffinityPropagation):
```

Comment:
could you explain this change?

Reply:
In newer versions of sk-learn, the `affinity` argument on  BaseEstimator is no longer available, yielding testing errors. It is, hower, available on AffinityPropagation, which uses BaseEstimator and ClusterMixin as parent classes. 

BaseEstimator and ClusterMixin are removed here because they're already included via AffinityPropagation.

Reply:
I think this can be resolved another way though, this documentation (https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html#sklearn.cluster.AgglomerativeClustering) indicates that the `metric` argument is probably what we should use when building an AgglomorativeClustering instance.

I'll update the PR with a less-intrusive change

Reply:
thanks! yeah i was just looking into this and it looks like the main thing was just that kwarg change